src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -271,33 +266,34 @@ export class KycService {
     return KycStepMapper.toKycResult(kycStep);
   }
 
-  async updateNationalityData(kycHash: string, stepId: number, data: KycNationalityData): Promise<KycResultDto> {
+  async updateUserData(
+    kycHash: string,
+    stepId: number,
+    data: Partial<UserData>,
+    requiresInternalReview: boolean,
+  ): Promise<KycResultDto> {
     let user = await this.getUser(kycHash);
     const kycStep = user.getPendingStepOrThrow(stepId);
 
-    const updatedUser = await this.userDataService.updateUserDataInternal(user, data);
+    user = await this.userDataService.updateUserDataInternal(user, data);
 
-    user = updatedUser.internalReviewStep(kycStep);
+    user = requiresInternalReview ? user.internalReviewStep(kycStep) : user.completeStep(kycStep, data);
```

Comment:
I think we can use the data also for internalReview `internalReviewStep(kycStep, data)`